Jandex fails with NPE on Java 21

Also build tool uses an older version that incorrectly tries to rebuild a newer index.

Stacktrace:
```
Caused by: java.lang.NullPointerException: Cannot read the array length because "bytes" is null
    at org.jboss.jandex.IndexWriterV2.writeByteTable (IndexWriterV2.java:228)
    at org.jboss.jandex.IndexWriterV2.write (IndexWriterV2.java:194)
    at org.jboss.jandex.IndexWriter.write (IndexWriter.java:107)
    at org.jboss.jandex.IndexWriter.write (IndexWriter.java:74)
    at io.helidon.build.linker.Jar.addIndex (Jar.java:567)
    at io.helidon.build.linker.Jar.copy (Jar.java:503)
    at io.helidon.build.linker.Jar.copyToDirectory (Jar.java:364)
    at io.helidon.build.linker.Application.lambda$install$1 (Application.java:102)
```

Related to https://github.com/helidon-io/helidon/issues/7221

--- a/linker/pom.xml
+++ b/linker/pom.xml
@@ -1,6 +1,6 @@
 <!--
 
-    Copyright (c) 2019, 2022 Oracle and/or its affiliates.
+    Copyright (c) 2019, 2023 Oracle and/or its affiliates.
 
     Licensed under the Apache License, Version 2.0 (the "License");
     you may not use this file except in compliance with the License.
@@ -42,7 +42,7 @@
             <version>${project.version}</version>
         </dependency>
         <dependency>
-            <groupId>org.jboss</groupId>
+            <groupId>io.smallrye</groupId>
             <artifactId>jandex</artifactId>
         </dependency>
         <dependency>

--- a/linker/src/main/java/module-info.java
+++ b/linker/src/main/java/module-info.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019, 2022 Oracle and/or its affiliates.
+ * Copyright (c) 2019, 2023 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -25,7 +25,7 @@ module helidon.linker {
 
     requires jdk.jlink;
     requires jdk.jdeps;
-    requires jandex;
+    requires org.jboss.jandex;
     requires org.fusesource.jansi;
     requires org.objectweb.asm;
     requires io.helidon.build.common;

--- a/pom.xml
+++ b/pom.xml
@@ -133,7 +133,7 @@
         <version.lib.hamcrest>1.3</version.lib.hamcrest>
         <version.lib.invoker>3.0.1</version.lib.invoker>
         <version.lib.jackson>2.11.0</version.lib.jackson>
-        <version.lib.jandex>2.1.1.Final</version.lib.jandex>
+        <version.lib.jandex>3.1.2</version.lib.jandex>
         <version.lib.jansi>1.18</version.lib.jansi>
         <version.lib.jaxb-api>2.3.3</version.lib.jaxb-api>
         <version.lib.jaxb-core>2.3.0.1</version.lib.jaxb-core>
@@ -808,7 +808,7 @@
                 <version>${version.lib.mustache}</version>
             </dependency>
             <dependency>
-                <groupId>org.jboss</groupId>
+                <groupId>io.smallrye</groupId>
                 <artifactId>jandex</artifactId>
                 <version>${version.lib.jandex}</version>
             </dependency>

